packages/inertia/src/router.ts

```diff
@@ -90,10 +97,14 @@ export class Router {
   }
 
   protected restoreScrollPositions(): void {
+    const canScrollTo = typeof document.documentElement.scrollTo === 'function'
     if (this.page.scrollRegions) {
       this.scrollRegions().forEach((region: Element, index: number) => {
         const scrollPosition = this.page.scrollRegions[index]
-        if (scrollPosition) {
+        if (scrollPosition && canScrollTo) {
+          region.scrollTo({ top: scrollPosition.top, left: scrollPosition.left })
+        }
+        else if (scrollPosition) {
           region.scrollTop = scrollPosition.top
           region.scrollLeft = scrollPosition.left
         }
```

Comment:
This might be a bit nicer to read
```suggestion
        if (! scrollPosition) {
          return
        }

        if (canScrollTo) {
          region.scrollTo(scrollPosition.top, scrollPosition.left)
        } else {
          region.scrollTop = scrollPosition.top
          region.scrollLeft = scrollPosition.left
        }
```